content/en/docs/tasks/run-application/horizontal-pod-autoscale.md

```diff
@@ -81,6 +82,86 @@ by using the scale sub-resource. Scale is an interface that allows you to dynami
 each of their current states. More details on scale sub-resource can be found
 [here](https://git.k8s.io/community/contributors/design-proposals/autoscaling/horizontal-pod-autoscaler.md#scale-subresource).
 
+### Algorithm Details
+
+From the most basic perspective, the Horizontal Pod Autoscaler controller
+operates on the ratio between desired metric value and current metric
+value:
+
+```
+desiredReplicas = ceil[currentReplicas * ( currentMetricValue / desiredMetricValue )]
+```
+
+For example, if the current metric value is `200m`, and the desired value
+is `100m`, the number of replicas will be doubled, since `float(200 / 100)
+== 2.0` If the the current value is instead `50m`, we'll halve the number
+of replicas, since `float(50 / 100) == 0.5`.  We'll skip scaling if the
+ratio is sufficiently close to 1.0 (within a globally-configurable
+tolerance, from the `--horizontal-pod-autoscaler-tolerance` flag, which
+defaults to 0.1).
+
+When a `targetAverageValue` or `targetAverageUtilization` is specified,
+the `currentMetricValue` is computed by taking the average of the given
+metric across all pods in the HorizontalPodAutoscaler's scale target.
+Before checking the tolerance and deciding on the final values, we take
+pod readiness and missing metrics into consideration, however.
+
+All pods with a deletion timestamp set (i.e. pods in the process of being
+shut down) and all failed pods are discarded.
+
+If a particular pod is missing metrics, it is set aside for later; pods
+with missing metrics will be used to adjust the final scaling amount.
+
+When scaling on CPU, if any pod has yet to become ready (i.e. it's still
+initializing) *or* the most recent metric point for the pod was before it
+became ready, that pod is set aside as well.
+
+Due to technical constraints, the HorizontalPodAutoscaler controller
```

Comment:
Nit: it might be useful to state explicitly that his paragraph still applies only to CPU-based scaling.